protostar/formatter/formatting_summary.py

```diff
@@ -23,30 +26,44 @@ def extend(self, formatting_result: FormattingResult):
         elif isinstance(formatting_result, IncorrectFormattingResult):
             self.incorrect.append(formatting_result)
 
-    def get_file_count(self):
+    def __len__(self) -> int:
         return len(self.broken) + len(self.correct) + len(self.incorrect)
 
     def any_unformatted_or_broken(self, check: bool):
         return bool(len(self.broken) + (len(self.incorrect) if check else 0))
 
+    def format_human(self, fmt: LogColorProvider) -> str:
+        total = len(self)
+        if total == 0:
+            return "No files found"
+
+        broken = len(self.broken)
+        incorrect = len(self.incorrect)
 
-def format_summary(summary: FormattingSummary, check: bool) -> str:
-    total = summary.get_file_count()
-    broken = len(summary.broken)
-    incorrect = len(summary.incorrect)
+        result: List[str] = [fmt.bold("Summary: ")]
 
-    result: List[str] = [log_color_provider.bold("Summary: ")]
+        text = "unformatted" if self.check else "reformatted"
+        color = "YELLOW" if self.check else "GREEN"
 
-    text = "unformatted" if check else "reformatted"
-    color = "YELLOW" if check else "GREEN"
+        result.append(fmt.colorize(color, f"{incorrect} {text}"))
 
-    result.append(log_color_provider.colorize(color, f"{incorrect} {text}"))
+        if broken:
+            result.append(", ")
+            result.append(fmt.colorize("RED", f"{broken} broken"))
 
-    if broken:
         result.append(", ")
-        result.append(log_color_provider.colorize("RED", f"{broken} broken"))
+        result.append(f"{total} total")
 
-    result.append(", ")
-    result.append(f"{total} total")
+        return "".join(result)
 
-    return "".join(result)
+    def format_object(self) -> dict:
+        return {
+            "type": "format",
+            "check": self.check,
+            "summary": {
+                "total": len(self),
+                "unchanged": len(self.correct),
+                "formatted": len(self.incorrect),
+                "broken": len(self.broken),
+            },
+        }
```

Comment:
Why is `"type": "format"` needed? I don't see why Protostar duplicates input data.

Reply:
https://github.com/software-mansion/protostar/discussions/1123